bzt/utils.py

```diff
@@ -340,28 +340,34 @@ def shell_exec(args, cwd=None, stdout=PIPE, stderr=PIPE, stdin=PIPE, shell=False
                      preexec_fn=os.setpgrp, close_fds=True, cwd=cwd, shell=shell, env=env)
 
 
-def readlines(_file, hint=None):
-    # get generator instead of list (in regular readlines())
-    length = 0
-    for line in _file:
-        yield line
-        if hint and hint > 0:
-            length += len(line)
-            if length >= hint:
-                return
-
-
 class FileReader(object):
-    def __init__(self, filename='', file_opener=open, parent_logger=logging.getLogger('')):
+    def __init__(self, filename="", file_opener=None, parent_logger=None):
         self.fds = None
+        if parent_logger:
+            self.log = parent_logger.getChild(self.__class__.__name__)
+        else:
+            self.log = logging.getLogger(self.__class__.__name__)
+
+        if file_opener:
+            self.file_opener = file_opener  # external method for opening of file
+        else:
+            self.file_opener = lambda f: open(f, mode='rb')     # default mode is binary
 
         # for non-trivial openers filename must be empty (more complicate than just open())
-        #  it turns all regular file checks off, see is_ready()
+        # it turns all regular file checks off, see is_ready()
         self.name = filename
-
-        self.file_opener = file_opener  # external method for opening of file
+        self.cp = 'utf8'    # default code page is utf8
```

Comment:
Note that Python usually spell that as `utf-8` (with a dash).

I did some research, and `https://docs.python.org/3/library/codecs.html#standard-encodings` says that `utf8` is recognized as a valid alias to `utf-8`, but only on CPython, not PyPy.